docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,350 @@
+both parsed with yyyy-MM-dd'T'hh:mm:ssX
+
+2010-01-01T01:02:03+01:02
+yyyy-MM-dd'T'hh:mm:ssXXX
+
+2010-01-01T01:02:03+01:02:03
+yyyy-MM-dd'T'hh:mm:ssXXXXX
+```
+
+
+If you expect all these combination to occur in your data you need to combine these patterns in 7.0
+[source,txt]
+--------------------------------------------------
+yyyy-MM-dd'T'hh:mm:ssX||yyyy-MM-dd'T'hh:mm:ssXXX||yyyy-MM-dd'T'hh:mm:ssXXXXX
+--------------------------------------------------
+
+
+The same applies if you expect your pattern to occur without a colon:
+Pattern `YYYY-MM-dd'T'hh:mm:ssZ` accepting these dates:
+```
+2010-01-01T01:02:03Z
+2010-01-01T01:02:03+01
+2010-01-01T01:02:03+0102
+2010-01-01T01:02:03+010203
+```
+should be migrated to
+[source,txt]
+--------------------------------------------------
+yyyy-MM-dd'T'hh:mm:ssX||yyyy-MM-dd'T'hh:mm:ssXX||yyyy-MM-dd'T'hh:mm:ssXXXX
+--------------------------------------------------
+--
+
+
+
+`d` day::
++
+--
+The same meaning in 7.0 but is less flexible.
+
+If your pattern was `YYYY-MM-dd` in 6.8 and you expect dates in form `2010-01-01` or `2010-01-1`
+You need to provide an alternative pattern for all of these `yyyy-MM-dd||yyyy-MM-d`.
+More then 2 digits are not allowed anymore. If you want to parse dates like `2010-01-00001` you need to create a pattern that expects a text literal `'000'` before `'d'`
+`yyyy-MM-'000'dd`
+--
+
+
+`e` name of day::
++
+--
+The same meaning in 7.0. However it is less flexible in 7.0. It do not allow parsing short and full text forms.
+
+If you used `EEE YYYY-MM` for a format and expected days in form:
+`Wed 2020-01` or `Wednesday 2020-01`
+then you have to use two combined patterns in 7.0
+`cccc yyyy-MM||ccc yyyy-MM`
+
+E vs c. E means day-of-week vs c is for localized day-of-week. The difference is that E won't behave correctly for full text forms like Wednesday
+--
+
+Text forms EEEE and similar::
++
+--
+Full text forms are depending on locale data provided with JDK and the implementation details of java vs joda. You should test carefully before upgrading these patterns.
+--
+
+`z` time zone text::
++
+--
+In 7.0 it will print 'Z' for Zulu given UTC timezone.
+--
+
+
+===== Test with your data
+All of the examples above should be tested with real data. It is likely that you do not need the flexibility joda was giving.
+If you expect your timezone to only be in one form - for instance `+01:00`  (XXX in java, ZZ in joda) -
+then there is no need to create a new java pattern with so many alternatives.
+
+Consider using this date debugging site for assistance https://esddd.herokuapp.com/
+
+==== Migrating affected mappings
+Once an index is created with a mapping, it cannot change already existing fields. You need to create a new index with updated mapping
+and reindex your data to it.
+You can however update your pipelines or templates. Remember to look for templates you use with tools outside ES.
+If you specified a custom date format there, then you need to update it too.
```

Comment:
```suggestion

[TIP]
====
When updating custom date formats, ensure you also update any templates or tools
used _outside_ of {es}.
====

You can update custom date formats in existing ingest pipelines and index
templates.

However, you cannot directly update date formats used in existing index
mappings. To change date formats in mappings, you must create a new index with
an updated mapping and reindex your data to it.
```

Reply:
I applied the suggestion on the sentence about the need to create a new mapping.
I think it is worth mentioning pipelines and templates here. This is about the the ones within ES itself (not external)